Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+          function (map: Dictionary<any>, itemStatus)
+          {
+            map[itemStatus.ID] = itemStatus;
+            return map;
+          },
+          {});
+
+        let previousVisibleItemIsSeparator = false;
+        for (var i = itemInfos.length - 1; i >= 0; i--)
+        {
+          const itemInfo = itemInfos[i];
+          const hasFilterOption = itemInfo.ID !== null;
+          const isSeparator = !hasFilterOption && itemInfo.Text === '-';
+          if (hasFilterOption)
+          {
+            const itemStatus = itemStatusMap[itemInfo.ID];
+            const isVisible = itemStatus !== undefined;
+            if (isVisible)
+            {
+              itemInfo.IsDisabled = itemStatus.IsDisabled;
+              previousVisibleItemIsSeparator = false;
+            }
+            else
+            {
+              itemInfos.splice (i, 1);
+            }
+          }
+          else if (isSeparator)
+          {
+            if (previousVisibleItemIsSeparator)
+              itemInfos.splice(i, 1);
+
+            previousVisibleItemIsSeparator = true;
+          }
+          else
+          {
+            previousVisibleItemIsSeparator = false;
+          }
+        }
+
+        onSuccess (itemInfos);
+      };
+
+      DropDownMenu._currentLoadOperation = currentLoadOperationID;
+      loadMenuItemStatus (onSuccessForLoadMenuItemStatus, onError);
+    }
+    else
+    {
+      onSuccess (itemInfos);
+    }
+  }
+
+  private static OnClick(context: JQuery, menuID: string, getSelectionCount: Nullable<() => number>, evt: Nullable<JQueryMouseEventObject>): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('context', context);
+
+    $('#' + menuID).data(DropDownMenu._button_timestampDataKey, new Date().getTime());
+
+    if (DropDownMenu._itemClicked)
+    {
+      DropDownMenu._itemClicked = false;
+      return;
+    }
+    if (DropDownMenu._currentMenu != null && DropDownMenu._currentMenu[0] !== context[0])
+    {
+      DropDownMenu.ClosePopUp(!DropDownMenu._updateFocus);
+    }
+    if (DropDownMenu._currentMenu == null)
+    {
+      var event: Nullable<JQueryMouseEventObject> = null;
+      if (evt != null)
+      {
+        var bounds = context[0].getBoundingClientRect();
+        if (bounds.left <= evt.clientX &&
+          bounds.right >= evt.clientX &&
+          bounds.top <= evt.clientY &&
+          bounds.bottom >= evt.clientY)
+          event = evt;
+      }
+      DropDownMenu.OpenPopUp(menuID, context, getSelectionCount, event!);
+      DropDownMenu._currentMenu = context;
+    }
+  }
+
+  private static OpenPopUp (menuID: string, context: JQuery, getSelectionCount: Nullable<() => number>, evt: Nullable<JQueryMouseEventObject>): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('context', context);
+
+    let selectionCount = -1;
+    if (getSelectionCount !== null)
+      selectionCount = getSelectionCount();
+
+    const menuInfo = DropDownMenu._menuInfos[menuID] as DropDownMenu_MenuInfo;
```

Comment:
```suggestion
    const menuInfo = DropDownMenu._menuInfos[menuID]!;
```

Prefer the bang over an `as` cast since. The `as` cast can be bad if the type changes because it can do stuff like: 

```ts
declare var test : number | string;
let a = test as string;
// a is now a string and number is gone
```